fabric-convention-tags-v2/src/datagen/java/net/fabricmc/fabric/impl/tag/convention/datagen/generators/ItemTagGenerator.java

```diff
@@ -0,0 +1,429 @@
+				.addOptionalTag(ConventionalItemTags.GRAY_DYES)
+				.addOptionalTag(ConventionalItemTags.PURPLE_DYES);
+		getOrCreateTagBuilder(ConventionalItemTags.BLACK_DYES)
+				.add(Items.BLACK_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.BLUE_DYES)
+				.add(Items.BLUE_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.BROWN_DYES)
+				.add(Items.BROWN_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.GREEN_DYES)
+				.add(Items.GREEN_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.RED_DYES)
+				.add(Items.RED_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.WHITE_DYES)
+				.add(Items.WHITE_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.YELLOW_DYES)
+				.add(Items.YELLOW_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.LIGHT_BLUE_DYES)
+				.add(Items.LIGHT_BLUE_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.LIGHT_GRAY_DYES)
+				.add(Items.LIGHT_GRAY_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.LIME_DYES)
+				.add(Items.LIME_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.MAGENTA_DYES)
+				.add(Items.MAGENTA_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.ORANGE_DYES)
+				.add(Items.ORANGE_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.PINK_DYES)
+				.add(Items.PINK_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.CYAN_DYES)
+				.add(Items.CYAN_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.GRAY_DYES)
+				.add(Items.GRAY_DYE);
+		getOrCreateTagBuilder(ConventionalItemTags.PURPLE_DYES)
+				.add(Items.PURPLE_DYE);
+	}
+
+	private void generateConsumableTags() {
+		Registries.ITEM.forEach(item -> {
+			if (item.getFoodComponent() != null) {
+				getOrCreateTagBuilder(ConventionalItemTags.FOODS).add(item);
+			}
+		});
+		getOrCreateTagBuilder(ConventionalItemTags.POTIONS)
+				.add(Items.LINGERING_POTION)
+				.add(Items.SPLASH_POTION)
+				.add(Items.POTION);
+	}
+
+	private void generateBucketTags() {
+		getOrCreateTagBuilder(ConventionalItemTags.EMPTY_BUCKETS)
+				.add(Items.BUCKET);
+		getOrCreateTagBuilder(ConventionalItemTags.LAVA_BUCKETS)
+				.add(Items.LAVA_BUCKET);
+		getOrCreateTagBuilder(ConventionalItemTags.ENTITY_WATER_BUCKETS)
+				.add(Items.AXOLOTL_BUCKET)
+				.add(Items.COD_BUCKET)
+				.add(Items.PUFFERFISH_BUCKET)
+				.add(Items.TROPICAL_FISH_BUCKET)
+				.add(Items.SALMON_BUCKET)
+				.add(Items.TADPOLE_BUCKET);
+		getOrCreateTagBuilder(ConventionalItemTags.WATER_BUCKETS)
+				.add(Items.WATER_BUCKET);
+		getOrCreateTagBuilder(ConventionalItemTags.MILK_BUCKETS)
+				.add(Items.MILK_BUCKET);
+	}
+
+	private void generateOreAndRelatedTags() {
+		// Categories
+		getOrCreateTagBuilder(ConventionalItemTags.DUSTS)
+				.add(Items.GLOWSTONE_DUST)
+				.add(Items.REDSTONE);
+		getOrCreateTagBuilder(ConventionalItemTags.GEMS)
+				.add(Items.DIAMOND, Items.EMERALD, Items.AMETHYST_SHARD, Items.LAPIS_LAZULI);
+		getOrCreateTagBuilder(ConventionalItemTags.INGOTS)
+				.add(Items.COPPER_INGOT, Items.GOLD_INGOT, Items.IRON_INGOT, Items.NETHERITE_INGOT);
```

Comment:
Metals should be ordered consistently. It seems like in most other places the ordering is `iron, copper, gold, netherite`.